x-pack/plugins/transform/public/app/sections/create_transform/components/step_define/step_define_form.tsx

```diff
@@ -500,459 +163,204 @@ export const StepDefineForm: FC<Props> = React.memo(({ overrides = {}, onChange,
-                          setAdvancedPivotEditorApplyButtonEnabled(true);
-                        } catch (e) {
-                          setAdvancedPivotEditorApplyButtonEnabled(false);
-                        }
-                      }}
-                      setOptions={{
-                        fontSize: '12px',
-                      }}
-                      theme="textmate"
-                      aria-label={i18n.translate(
-                        'xpack.transform.stepDefineForm.advancedEditorAriaLabel',
-                        {
-                          defaultMessage: 'Advanced pivot editor',
-                        }
-                      )}
-                    />
-                  </EuiPanel>
-                </EuiFormRow>
-              </Fragment>
+              <AdvancedPivotEditor {...stepDefineForm.advancedPivotEditor} />
             )}
-            <EuiFormRow>
-              <EuiFlexGroup gutterSize="none">
-                <EuiFlexItem>
-                  <EuiSwitch
-                    label={i18n.translate(
-                      'xpack.transform.stepDefineForm.advancedEditorSwitchLabel',
-                      {
-                        defaultMessage: 'Advanced pivot editor',
-                      }
-                    )}
-                    checked={isAdvancedPivotEditorEnabled}
-                    onChange={() => {
-                      if (
-                        isAdvancedPivotEditorEnabled &&
-                        (isAdvancedPivotEditorApplyButtonEnabled ||
-                          advancedEditorConfig !== advancedEditorConfigLastApplied)
-                      ) {
-                        setAdvancedEditorSwitchModalVisible(true);
-                        return;
-                      }
-
-                      toggleAdvancedEditor();
-                    }}
-                    data-test-subj="transformAdvancedPivotEditorSwitch"
-                  />
-                  {isAdvancedEditorSwitchModalVisible && (
-                    <SwitchModal
-                      onCancel={() => setAdvancedEditorSwitchModalVisible(false)}
-                      onConfirm={() => {
-                        setAdvancedEditorSwitchModalVisible(false);
-                        toggleAdvancedEditor();
-                      }}
-                      type={'pivot'}
-                    />
-                  )}
-                </EuiFlexItem>
-                {isAdvancedPivotEditorEnabled && (
+          </EuiFlexItem>
+          <EuiFlexItem grow={false} style={{ width: advancedEditorsSidebarWidth }}>
+            <EuiFlexGroup gutterSize="xs" direction="column" justifyContent="spaceBetween">
+              <EuiFlexItem grow={false}>
+                <EuiFormRow hasEmptyLabelSpace>
+                  <EuiFlexGroup alignItems="center" justifyContent="spaceBetween">
+                    <EuiFlexItem grow={false}>
+                      <AdvancedPivotEditorSwitch {...stepDefineForm} />
```

Comment:
Could the switch be moved to the top of this section? Currently it is aligned with the 'Group by' control which gives the impression it is only for editing the `group_by` part of the pivot.

Reply:
As discussed, I'll leave it as is for now, moving it up looks misaligned (see screenshot), moving it even further up would result in additional vertical whitespace 
![image](https://user-images.githubusercontent.com/230104/80955296-562a2180-8dff-11ea-835b-3b860a705772.png)